src/configuration/app/workers.md

```diff
@@ -0,0 +1,209 @@
+        mounts:
+            "scratch": "shared:files/scratch"
+
+        
+    mail:
+        size: 'S'
+        commands:
+            start: |
+                python mail-worker.py
+        variables:
+            env:
+                type: 'worker'
+        disk: 256
+        mounts: {}
+        relationships:
+            emails: 'rabbitqueue:rabbitmq'
+```
+
+There's a lot going on here, but it's all reasonably straightforward.  This configuration will take a single Python 3.6 code base from your repository, download all dependencies in `requirements.txt`, and the install Gunicorn.  That artifact (your code plus the downloaded dependencies) will be deployed as three separate container instances, all running Python 3.6.
+
+The `web` instance will start a gunicorn process to serve a web application.  
+
+* It will run the gunicorn process to serve web requests, defined by the `project/wsgi.py` file which contains an `application`definition. 
+* It will have an environment variable named `TYPE` with value `web`.
+* It will have a writeable mount at `/app/uploads` with a maximum space of 2048 MB.
+* It will have access to both a MySQL database and a RabbitMQ server, both of which are defined in `services.yaml`.
+* Platform.sh will automatically allocate resources to it as available on the plan, once all fixed-size containers are allocated.
+
+The `queue` instance will be a worker that is not web-accessible.
+
+* It will run the `queue-worker.py` script, and restart it automatically if it ever terminates.
+* It will have an environment variable named `TYPE` with value `worker`.
+* It will have a writeable mount at `/app/scratch` with a maximum space of 512 MB.
+* It will have access to both a MySQL database and a RabbitMQ server, both of which are defined in `services.yaml` (because it doesn't specify otherwise).
+* It will have "Medium" levels of CPU and RAM allocated to it, always.
+
+The `mail` instance will be a worker that is not web-accessible.
+
+* It will run the `mail-worker.py` script, and restart it automatically if it ever terminates.
+* It will have an environment variable named `TYPE` with value `worker`.
+* It will have no writeable file mounts at all.
+* It will have access only to the RabbitMQ server, through a different relationship name than on the `web` instance.  It will have no access to MySQL whatsoever.
+* It will have "Small" levels of CPU and RAM allocated to it, always.
+
+This way, the web instance has a large upload space, the queue instance has a small amount of scratch space for temporary files, and the mail instance has no writeable disk space at all as it doesn't need it.  The mail instance also does not need any access to the SQL database so for security reasons it has none.  The workers have known fixed sizes, while web can scale to as large as the plan allows.  Each instance can also check the `TYPE` environment variable to detect how it's running and, if appropriate, vary its behavior accordingly.
```

Comment:
Maybe this should say "the mail instance has no persistent writeable disk space"? `/tmp` is still available.